### PR TITLE
smassh: init at 3.1.3

### PR DESCRIPTION
**Pull request by AimPizza**
## Description of changes

[Smassh](https://github.com/kraanzu/smassh) is a TUI based typing test application inspired by MonkeyType 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/smassh/default.nix
Comment (lovesegfault) [inline pkgs/tools/misc/smassh/default.nix:1]: This file should go in `pkgs/by-name` instead

Comment (AimPizza) [reply pkgs/tools/misc/smassh/default.nix:1]: I truly am sorry for the many oversights. It should work now, though.

**Comment by lovesegfault** [comment]
Please rebase/squash your changes, if possible

File: pkgs/by-name/sm/smassh/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  passthru.tests.version = testers.testVersion {
+    package = smassh;
+    command = "HOME=$(mktemp -d) smassh --version";
+  };
+
+  meta = with lib; {
+    description = "a TUI based typing test application inspired by MonkeyType";
+    homepage = "https://github.com/kraanzu/smassh";
+    changelog = "https://github.com/kraanzu/smassh/blob/main/CHANGELOG.md";
+    license = licenses.gpl3;
```
Comment (MikaelFangel) [inline pkgs/by-name/sm/smassh/package.nix:48]: The licenses should be either gpl3Plus or gpl3Only depending on which one matches.

gpl3 is deprecated. See [https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206](https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206)

File: pkgs/by-name/sm/smassh/package.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  # No tests available
+  doCheck = false;
+
+  passthru.tests.version = testers.testVersion {
+    package = smassh;
+    command = "HOME=$(mktemp -d) smassh --version";
+  };
+
+  meta = with lib; {
+    description = "a TUI based typing test application inspired by MonkeyType";
```
Comment (MikaelFangel) [inline pkgs/by-name/sm/smassh/package.nix:45]:
```suggestion
    description = "A TUI based typing test application inspired by MonkeyType";
```

**Comment by MikaelFangel** [comment]
A technicality but the commit adding you as a maintainer needs to come before the package commit (:

**Comment by AimPizza** [comment]
We'll get there eventually 😁
Thank you everybody being so patient and helpful!

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -649,6 +649,12 @@
       fingerprint = "B0D7 2955 235F 6AB5 ACFA  1619 8C7F F5BB 1ADE F191";
     }];
   };
+  aimpizza = {
+    email = "rickomo.us@gmail.com";
+    name = "Rick Omonsky";
+    github = "AimPizza";
+    githubId = "64905268";
```
Comment (MikaelFangel) [inline maintainers/maintainer-list.nix:656]:
```suggestion
    githubId = 64905268;
```

Comment (AimPizza) [reply maintainers/maintainer-list.nix]: What I do wonder though, is squashing and force-pushing after every change the correct approach?

Comment (lovesegfault) [reply maintainers/maintainer-list.nix]: Yeah, that's the usual workflow :)

**Pull request merged**